### PR TITLE
feat(transfers): keep key subset at replicas on section split

### PR DESCRIPTION
**Pull request by oetyng**
- When section splits, the Replicas in either resulting section
also split the responsibility of the accounts.
Thus, both Replica groups need to drop the accounts that
the other group is now responsible for.

### CODE REVIEW

File: src/node/elder_duties/data_section/mod.rs [hunk 1]
```diff
@@ -84,6 +84,11 @@ impl DataSection {
         self.rewards.transition(actor)
     }
 
+    // At section split, all Elders get their reward payout.
+    pub fn section_split(&mut self) -> Option<NodeOperation> {
+        None
```
Comment (Yoga07) [inline src/node/elder_duties/data_section/mod.rs:89]: Would be nice to add a TODO/FIXME stating to fix this once we have section splits?

File: src/node/elder_duties/key_section/transfers/replica_manager.rs [hunk 2]
```diff
@@ -82,6 +87,28 @@ impl ReplicaManager {
...
+        self.check_init_status()?;
+
+        // Drops the streams from db.
+        self.store
+            .drop(accounts)
+            .map_err(|e| NdError::NetworkOther(e.to_string()))?;
+
+        // Replays the kept streams
+        // on a new instance of a Replica.
+        self.churn(
```
Comment (Yoga07) [inline src/node/elder_duties/key_section/transfers/replica_manager.rs:104]: Makes sense to create a new one here, but should we necessarily? Won't dropping the non-matching ones be sufficient?

File: src/node/elder_duties/data_section/mod.rs (hunk 1 above)
Comment (oetyng) [reply src/node/elder_duties/data_section/mod.rs:89]: I see what you mean, but [my upcoming PR has code in this method](https://github.com/oetyng/safe-vault/blob/8585fc1516e76a3a7ea874c5962640497856fccf/src/node/elder_duties/data_section/mod.rs#L90).

File: src/node/elder_duties/key_section/transfers/replica_manager.rs (hunk 2 above)
Comment (oetyng) [reply src/node/elder_duties/key_section/transfers/replica_manager.rs:104]: What we drop is from the database, but the replica in our field is a materialized view from the events in the database, so we need to reload it, or it is essentially like keeping a stale cache.

Comment (Yoga07) [reply src/node/elder_duties/key_section/transfers/replica_manager.rs:104]: Right, better to create a new one :)

**Pull request merged**